openflow15/nxt_message.go

```diff
@@ -775,14 +773,17 @@ type PacketIn2PropPacket struct {
 }
 
 func (p *PacketIn2PropPacket) Len() (n uint16) {
-	n = p.PropHeader.Len() + p.Packet.Len()
+	return uint16(p.actLength())
+}
 
+func (p *PacketIn2PropPacket) actLength() int {
+	n := int(p.PropHeader.Len()) + int(p.Packet.Len())
 	// Round it to closest multiple of 8
 	return ((n + 7) / 8) * 8
 }
 
 func (p *PacketIn2PropPacket) MarshalBinary() (data []byte, err error) {
-	data = make([]byte, p.Len())
+	data = make([]byte, p.actLength())
```

Comment:
out of curiosity, do we ever marshal PacketIn messages with this library? We obviously unmarshal them when we receive them from OVS, but I don't expect that we ever marshal them?

Reply:
No, we don't marshal a packetIn. It is used in the test code. It may cause failure if we don't resolve it.